cmd/tink-worker/internal/registry.go

```diff
@@ -0,0 +1,71 @@
+package internal
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"io"
+	"os"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"github.com/packethost/pkg/log"
+	"github.com/pkg/errors"
+)
+
+// RegistryConnDetails are the connection details for accessing a Docker
+// registry and logging activities
+type RegistryConnDetails struct {
+	registry,
+	user,
+	pwd string
+	logger log.Logger
+}
+
+// NewRegistryConnDetails creates a new RegistryConnDetails
+func NewRegistryConnDetails(registry, user, pwd string, logger log.Logger) *RegistryConnDetails {
+	return &RegistryConnDetails{
+		registry: registry,
+		user:     user,
+		pwd:      pwd,
+		logger:   logger,
+	}
+}
+
+// NewClient uses the RegistryConnDetails to create a new Docker Client
+func (r *RegistryConnDetails) NewClient() (*client.Client, error) {
+	if r.registry == "" {
+		return nil, errors.New("required DOCKER_REGISTRY")
+	}
+	c, err := client.NewClientWithOpts(client.FromEnv, client.WithHost(r.registry), client.WithAPIVersionNegotiation())
```

Comment:
```suggestion
func (r *RegistryConnDetails) NewClient() (*client.Client, error) {
	c, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
```

Reply:
WithHost here is setting the docker daemon host rather than the registry url.

Reply:
Removed WithHost in the latest push (which has been rebased).